### ISSUE TITLE
VDiskController.extend uses truncate

### ISSUE THREAD

**Issue by khenderick** [openvstorage/framework#892]
VDiskController.extend uses truncate. It should use a volumedriver API

**Comment by khenderick** [comment]
Also expose it over the API.

**Comment by khenderick** [comment]
Fixed by #894, packaged in openvstorage-2.7.3-rev.3953.edd2b11

**Comment by JeffreyDevloo** [comment]
### Steps
- Installed OVH
- Setup disk roles
- Setup backend
- Setup vpool
- Added vdisk
- Used postman to send api call
  ![selection_081](https://cloud.githubusercontent.com/assets/17570109/18630891/49780fae-7e6f-11e6-9bd6-9f74c75fe414.png)
  ### Output
  Tested two cases: a smaller volume and a bigger volume

**Smaller volume:** 
Fetching task details:

```
{
  "status": "FAILURE",
  "successful": false,
  "failed": true,
  "result": "Shrinking is not possible",
  "ready": true,
  "id": "45372824-7749-457f-93bb-673b729e3aa0"
}
```

**Bigger volume:** 
Fetching task details:

```
{
  "status": "SUCCESS",
  "successful": true,
  "failed": false,
  "result": null,
  "ready": true,
  "id": "45dfe5b3-57e5-498e-84bd-8127492a584e"
}
```

Found the following in volume driver logging:

```
2016-09-19 13:42:18 888886 +0200 - ovs-node1 - 1604/0x00007f397effd700 - volumedriverfs/XMLRPCTimingWrapper - 0000000000002766 - info - execute: Arguments for resizeObjec
t are {[volume_id:d0445b8a-7a30-49b2-b57e-bf803c436f41,volume_size:3221225472B,vrouter_cluster_id:94d36c61-1951-44af-8fe6-8c2441b8d8ea]}
2016-09-19 13:42:18 891044 +0200 - ovs-node1 - 1604/0x00007f397effd700 - volumedriverfs/XMLRPCTimingWrapper - 0000000000002767 - info - execute: Call resizeObject took 0.
002092 seconds
```

Task is correctly processed.

**Verification on GUI:**:
- The old volume size was 2GB
- New size _should_ be 3GB

Found:
![selection_082](https://cloud.githubusercontent.com/assets/17570109/18631026/fd954e48-7e6f-11e6-8e58-fd44a5118207.png)

```
root@ovs-node2:~# ls -al /mnt/vm-vpool/test1.raw 
-rwxrwxr-x 1 ovs ovs 3221225472 Sep 19 11:57 /mnt/vm-vpool/test1.raw
```

**Test succeeded for this scope: however expected 3GB on GUI and got 2GB**

### Test result

Test succeed. However please note that the size was not displayed correctly on the GUI.

Hyperconverged setup
- Three nodes with each three disks for the back-end

### Package information
- openvstorage                       2.7.3-rev.3978.4f93e02-1                                 amd64        openvStorage
- openvstorage-backend               1.7.3-rev.709.5bb94c8-1                                  amd64        openvStorage Backend plugin
- openvstorage-backend-core          1.7.3-rev.709.5bb94c8-1                                  amd64        openvStorage Backend plugin core
- openvstorage-backend-webapps       1.7.3-rev.709.5bb94c8-1                                  amd64        openvStorage Backend plugin Web Applications
- openvstorage-cinder-plugin         1.2.2-rev.38.dcc3b76-1                                   amd64        OpenvStorage Cinder plugin for OpenStack
- openvstorage-core                  2.7.3-rev.3978.4f93e02-1                                 amd64        openvStorage core
- openvstorage-hc                    1.7.3-rev.709.5bb94c8-1                                  amd64        openvStorage Backend plugin HyperConverged
- openvstorage-health-check          2.0.0-rev.134.9ef41c0-1                                  amd64        Open vStorage HealthCheck
- openvstorage-sdm                   1.6.3-rev.357.ee7c9f5-1                                  amd64        Open vStorage Backend ASD Manager
- openvstorage-test                  2.7.3-rev.1012.608f550-1                                 amd64        openvStorage autotest suite
- openvstorage-webapps               2.7.3-rev.3978.4f93e02-1                                 amd64        openvStorage Web Applications
